src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -34,12 +36,14 @@ private sealed partial class SymbolStartAnalyzer
             public SymbolStartAnalyzer(
                 AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer compilationAnalyzer,
                 INamedTypeSymbol eventArgsTypeOpt,
-                ImmutableHashSet<INamedTypeSymbol> attributeSetForMethodsToIgnore)
+                ImmutableHashSet<INamedTypeSymbol> attributeSetForMethodsToIgnore,
+                Compilation compilation)
             {
                 _compilationAnalyzer = compilationAnalyzer;
 
                 _eventArgsTypeOpt = eventArgsTypeOpt;
                 _attributeSetForMethodsToIgnore = attributeSetForMethodsToIgnore;
+                _serializationConstructorCheck = new SerializationConstructorCheck(compilation);
```

Comment:
❓ Can we avoid looking these types up for analysis of _every symbol_, and instead do it once for the compilation?

Reply:
Good idea. Fixed in 7898c2d97af61d4161c1c7942d420dfad9d71fe8.